hazelcast-jet-core/src/main/java/com/hazelcast/jet/JetInstance.java

```diff
@@ -138,6 +141,17 @@ default Job newJob(@Nonnull Pipeline pipeline) {
      */
     @Nonnull
     default Job newJob(@Nonnull Pipeline pipeline, @Nonnull JobConfig config) {
+        PipelineImpl impl = (PipelineImpl) pipeline;
+        for (Entry<String, File> e : impl.attachedFiles().entrySet()) {
+            File file = e.getValue();
+            if (file.isDirectory()) {
+                config.attachDirectory(file, e.getKey());
+            } else if (file.isFile()) {
```

Comment:
`isFile` returns false if the file is in a directory which lacks the `x` attribute, see [here](https://stackoverflow.com/questions/25039156/java-what-constitutes-a-normal-file-in-the-context-of-file-isfile). It's better to use `canRead` or to not check at all and rely when we try to read it.

Reply:
`isFile` is used ubiquitously to check that a file exists as a regular file. I can only add one more check `file.canRead`

Reply:
You can't open a file in a directory without the `x` permission:

```
$ cat test/test.txt 
test
$ chmod -x test
$ cat test/test.txt 
cat: test/test.txt: Permission denied
```